build/proto/Dockerfile

```diff
@@ -32,23 +35,61 @@ ENV GOPRIVATE=$GOPRIVATE
 ARG PROTOC_VERSION
 ARG PROTOC_SHA1
 ARG MAVEN_URL
+ARG NUGET_URL
+
+RUN apt-get update -y
+RUN apt-get install -y unzip
+
 RUN curl -OLk ${MAVEN_URL}/com/google/protobuf/protoc/${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-x86_64.exe && \
     mv protoc-${PROTOC_VERSION}-linux-x86_64.exe /usr/local/bin/protoc && \
     echo "${PROTOC_SHA1} /usr/local/bin/protoc" | sha1sum -c - && \
     chmod +x /usr/local/bin/protoc
 
+# Install the csharp grpc plugin
+RUN curl -sSL https://www.nuget.org/api/v2/package/Grpc.Tools/2.46.3 --output grpctools.zip && \
+    unzip -q -d grpctools grpctools.zip && \
+    mv grpctools/tools/linux_x64/grpc_csharp_plugin /usr/local/bin/grpc_csharp_plugin && \
+    chmod +x /usr/local/bin/grpc_csharp_plugin
+
+# Remove after I copy proto files
 ARG GOGO_PROTOBUF_VERSION
-RUN GO111MODULE=on go get github.com/gogo/protobuf/protoc-gen-gogofaster@v${GOGO_PROTOBUF_VERSION}
+RUN GO111MODULE=on go get github.com/gogo/protobuf/protoc-gen-gogofaster@v${GOGO_PROTOBUF_VERSION} && \
```

Comment:
Let's talk on monday and figure out a way to avoid adding all this back in.  

https://github.com/G-Research/armada/pull/1083/files

I merged this PR to try and clean these docker files up so we don't need to manage the version of K8 files in docker files.  I really want a single place where we pull these docker files and not replicate it again and again for every proto generation.

Reply:
We are going to file this as a separate issue to cleanup later.